ranndy/models/auto_encoder.py

```diff
@@ -57,9 +58,11 @@ def _build_embedding(self, shape):
     def _build_encoder(self):
         # Build RNN cell
         self.encoder_cell = tf.nn.rnn_cell.LSTMCell(self.lstm_size)
-        self.encoder_outputs, self.encoder_state = tf.nn.dynamic_rnn(self.encoder_cell, self.enc_embedding_output,
-                                                                     sequence_length=self.iterator.source_length,
-                                                                     dtype=tf.float32)
+        _, encoder_state = tf.nn.dynamic_rnn(self.encoder_cell, self.enc_embedding_output,
+                                             sequence_length=self.iterator.source_length,
+                                             dtype=tf.float32)
+        # Assign encoder state as decoder state input
+        self.decoder_state_input = encoder_state
```

Comment:
Why add the extra step?

Reply:
This way the decoder can be re-used by the VAE. I could have re-used the self.encoder_state variable but it would have been deceiving in the case of the VAE (where the decoder input is a sampled sample) .